helpers/login.ts

```diff
@@ -13,47 +13,55 @@ const E2E_ACCOUNT_LOGIN = 'e2e_tester@shopcanal.com';
 
 /**
  * Use this to log a user in and make sure they land on a given page.
+ *
+ * @param loginUrl
+ * @param firstLoggedInPageUrl When passed, check that the user lands on this page.
  */
 const logIn = async (
   page: Page,
   context: BrowserContext,
   loginUrl: string,
-  firstLoggedInPageUrl: string,
+  firstLoggedInPageUrl: string | null = null,
 ): Promise<void> => {
   const loginButton = page.locator(LOGIN_BUTTON_SELECTOR);
   const profileDropdown = page.locator('button:has-text("e2e_tester")');
 
-  if (process.env.APP_TEST_PASSWORD) {
-    // Make sure browser is logged out before attempting to go through login flow
-    await logout(page, context);
+  if (!process.env.APP_TEST_PASSWORD) {
+    console.warn('Could not log in because no APP_TEST_PASSWORD was provided. Failing test.');
+    expect(true).toBe(false);
+  }
 
-    // Navigate to login page and wait for login button to load
-    await page.goto(loginUrl);
-    await loginButton.waitFor();
+  // Make sure browser is logged out before attempting to go through login flow
+  await logout(page, context);
 
-    // Fill out email and password
-    await page.fill(LOGIN_EMAIL_INPUT_SELECTOR, E2E_ACCOUNT_LOGIN);
-    await page.fill(LOGIN_PASSWORD_INPUT_SELECTOR, process.env.APP_TEST_PASSWORD || '');
+  // Navigate to login page and wait for login button to load
+  await page.goto(loginUrl);
+  await loginButton.waitFor();
 
-    // Then click the login button and make sure we navigate
-    await loginButton.click();
+  // Fill out email and password
+  await page.fill(LOGIN_EMAIL_INPUT_SELECTOR, E2E_ACCOUNT_LOGIN);
+  await page.fill(LOGIN_PASSWORD_INPUT_SELECTOR, process.env.APP_TEST_PASSWORD || '');
 
-    // Make sure the profile dropdown shows, as this appears on both sides once logged in
-    await profileDropdown.waitFor();
+  // Then click the login button and make sure we navigate
+  await loginButton.click();
 
-    // Make sure this is the page we want
-    expect(page.url().includes(firstLoggedInPageUrl)).toBeTruthy();
-  } else {
-    console.warn('Could not log in because no APP_TEST_PASSWORD was provided. Failing test.');
-    expect(true).toBe(false);
+  // Make sure the profile dropdown shows, as this appears on both sides once logged in
+  await profileDropdown.waitFor();
+
+  // Make sure we're no longer in the login
+  expect(page.url().includes(loginUrl)).toBeFalsy();
+
+  // Make sure this is the page we want
+  if (firstLoggedInPageUrl) {
+    expect(page.url().includes(firstLoggedInPageUrl)).not.toBeTruthy();
```

Comment:
why's this inverted? We want to test that the page we're on IS the page we were looking for

Reply:
Right! This is not ready for review yet, I'll ping you in a couple of hours? 🚑